File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/chunkloader/ChunkLoaderCreator.java
package com.gmail.borlandlp.minigamesdtools.arena.chunkloader;

import org.bukkit.Chunk;
import org.bukkit.Location;
import org.bukkit.World;

import java.util.ArrayList;
import java.util.List;

public class ChunkLoaderCreator {
    public ChunksLoader buildChunkLoader(World world, Location min_point, Location max_point) throws Exception {
        if(min_point.getX() > max_point.getX() || min_point.getZ() > max_point.getZ()) { // Swap locations if programmer was mistaken
            Location buff = min_point.clone();
            min_point = max_point;
            max_point = buff;
        }

        List<Chunk> chunks = new ArrayList<>();
        for (int x = min_point.getBlockX(); x <= max_point.getBlockX(); x += 16) {
            for (int z = min_point.getBlockZ(); z <= max_point.getBlockZ(); z += 16) {
                Chunk chunk = world.getChunkAt(x, z);
                if(chunk != null) {
                    chunks.add(chunk);
                } else {
                    throw new Exception("Error while parsing chunks for min_point:" + min_point + " # max_point:" + max_point);
                }
            }
        }

        ChunksLoader chunksLoader = new ChunksLoader();
        chunksLoader.setChunks(chunks);

        return chunksLoader;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/chunkloader/ChunkLoaderCreator.kt
package com.gmail.borlandlp.minigamesdtools.arena.chunkloader

import org.bukkit.Chunk
import org.bukkit.Location
import org.bukkit.World
import java.util.*

class ChunkLoaderCreator {
    @Throws(Exception::class)
    fun buildChunkLoader(
        world: World,
        minPoint: Location,
        maxPoint: Location
    ): ChunksLoader {
        val isRevertedLocations = minPoint.x > maxPoint.x || minPoint.z > maxPoint.z
        val minLoc: Location = if (isRevertedLocations) maxPoint else minPoint
        val maxLoc: Location = if (isRevertedLocations) minPoint else maxPoint

        val chunks: MutableList<Chunk> = mutableListOf()
        var x = minLoc.blockX
        while (x <= maxLoc.blockX) {
            var z = minLoc.blockZ
            while (z <= maxLoc.blockZ) {
                val chunk = world.getChunkAt(x, z)
                if (chunk != null) {
                    chunks.add(chunk)
                } else {
                    throw Exception("Error while parsing chunks for min_point:$minLoc # max_point:$maxLoc")
                }
                z += 16
            }
            x += 16
        }

        return ChunksLoader().apply {
            this.chunks = chunks
        }
    }
}